commons/src/main/java/org/infinispan/commons/util/ByRef.java

```diff
@@ -25,4 +25,71 @@ public T get() {
    public void set(T t) {
       ref = t;
    }
+
+	/**
+    * Implementation for primitive type
+    */
+   public static class Boolean {
+      boolean ref;
+
+      public Boolean(boolean b) {
+         ref = b;
+      }
+
+      public boolean get() {
+         return ref;
+      }
+
+      public void set(boolean b) {
+         this.ref = b;
+      }
+   }
+
+   public static class Pair<T1, T2> {
```

Comment:
The idea is that these are two values that are returned by reference (in order to reduce the number of arguments passed to the function and allocations required), while if I had a generic pair, I would prefer to have it immutable.